audit-service/src/unit-tests/java/uk/gov/gchq/palisade/service/audit/AbstractSerialisationTest.java

```diff
@@ -0,0 +1,127 @@
+
+    /**
+     * Tests the serialisation and deserialisation of each object provided by the
+     * method source
+     * <p>
+     * The tests asserted by this method:
+     * <ul>
+     * <li><em>Equals</em> - Test equality using the each instances equals
+     * method</li>
+     * <li><em>Recursive Equals</em> - Tests equality by testing field by field in
+     * the instance</li>
+     * </ul>
+     *
+     * @param type             The type class
+     * @param <O>              The type of the object itself
+     * @param <T>              The type of the object
+     * @param expectedInstance The expected instance
+     * @return The object being tested
+     * @throws Exception if an error occurs
+     */
+    protected <O, T extends O> T testInstance(final Class<T> type, final O expectedInstance) throws Exception {
+        return testInstance(type, expectedInstance, null);
+    }
+
+    /**
+     * Tests the serialisation and deserialisation of each object provided by the
+     * method source
+     * <p>
+     * The tests asserted by this method:
+     * <ul>
+     * <li><em>Equals</em> - Test equality using the each instances equals
+     * method</li>
+     * <li><em>Recursive Equals</em> - Tests equality by testing field by field in
+     * the instance</li>
+     * <li><em>JSON Attributes</em> - uses JSONAssert to test whether the serialise
+     * instance matches the provided JSON string</li>
+     * </ul>
+     *
+     * @param type             The type class
+     * @param <O>              The type of the object itself
+     * @param <T>              The type
+     * @param expectedInstance The expected instance
+     * @param expectedJson     The expected JSON of the provided instance. This JSON
+     *                         string is asserted of ignored is {@code expectedJson}
+     *                         is null.
+     * @return The object being tested
+     * @throws Exception if an error occurs
+     */
+    protected <O, T extends O> T testInstance(final Class<T> type, final O expectedInstance, final String expectedJson)
+        throws Exception {
+
+        var objectMapper = getObjectMapper();
+
+        var actualJson = objectMapper.writeValueAsString(expectedInstance);
+        var actualInstance = objectMapper.readValue(actualJson, type);
+
+        assertThat(actualInstance)
+            .as("Using toString(), the original %s is the same as the deserialised version",
+                expectedInstance.getClass().getSimpleName())
+            .isEqualTo(expectedInstance);
+
+        assertThat(actualInstance)
+            .as("Using recursive toString(), the original %s is the same as the deserialised version",
+                expectedInstance.getClass().getSimpleName())
+            .usingRecursiveComparison()
+            .isEqualTo(expectedInstance);
+
+        if (expectedJson != null && !expectedJson.isBlank()) {
+            assertEquals(expectedJson, actualJson, JSONCompareMode.STRICT);
+        }
```

Comment:
expectedJson is hardcoded to be null so this wont run

Reply:
There are two testXXX methods. One takes 3 parameters and one takes 2. The one with two does not require the json string. This is correct.